mission_ctl.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+func (r *regolancer) addFailedChan(fromStr string, toStr string, amount int64) {
+	r.mcCache[fromStr+toStr] = amount
+}
+
+func (r *regolancer) validateRoute(route *lnrpc.Route) error {
+	lastHopPK := ""
```

Comment:
why not initialize `lastHopPK = r.myPK` bc thats actually the "from" of the first channel

Reply:
Oops, forgot we have that!